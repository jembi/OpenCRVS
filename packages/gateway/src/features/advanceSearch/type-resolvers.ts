/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
 * graphic logo are (registered/a) trademark(s) of Plan International.
 */

import { GQLEvent } from '@gateway/graphql/schema'

export interface ISearch extends IAdvancedSearchParameters {
  name: string
}
export interface IAdvancedSearchParameters {
  event?: GQLEvent
  registrationStatuses?: string[]
  dateOfEvent?: string
  dateOfEventStart?: string
  dateOfEventEnd?: string
  registrationNumber?: string
  trackingId?: string
  dateOfRegistration?: string
  dateOfRegistrationStart?: string
  dateOfRegistrationEnd?: string
  declarationLocationId?: string
  declarationJurisdictionId?: string
  eventLocationId?: string
  eventLocationLevel1?: string
  eventLocationLevel2?: string
  eventLocationLevel3?: string
  eventLocationLevel4?: string
  eventLocationLevel5?: string
  childFirstNames?: string
  childLastName?: string
  childDoB?: string
  childDoBStart?: string
  childDoBEnd?: string
  childGender?: string
  deceasedFirstNames?: string
  deceasedFamilyName?: string
  deceasedGender?: string
  deceasedDoB?: string
  deceasedDoBStart?: string
  deceasedDoBEnd?: string
  motherFirstNames?: string
  motherFamilyName?: string
  motherDoB?: string
  motherDoBStart?: string
  motherDoBEnd?: string
  fatherFirstNames?: string
  fatherFamilyName?: string
  fatherDoB?: string
  fatherDoBStart?: string
  fatherDoBEnd?: string
  informantFirstNames?: string
  informantFamilyName?: string
  informantDoB?: string
  informantDoBStart?: string
  informantDoBEnd?: string
}

export interface IBookmarkAdvancedSearchPayload {
  userId: string
  search: ISearch
}
