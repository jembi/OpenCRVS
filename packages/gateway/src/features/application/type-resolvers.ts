/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */

import {
  GQLCurrencyInput,
  GQLBirthInput,
  GQLDeathInput,
  GQLLoginBackgroundInput,
  GQLCountryLogoInput,
  GQLMarriageInput
} from '@gateway/graphql/schema'

export interface IApplicationConfigPayload {
  APPLICATION_NAME: string
  BIRTH: GQLBirthInput
  COUNTRY_LOGO: GQLCountryLogoInput
  CURRENCY: GQLCurrencyInput
  DEATH: GQLDeathInput
  MARRIAGE: GQLMarriageInput
  FIELD_AGENT_AUDIT_LOCATIONS: string
  EXTERNAL_VALIDATION_WORKQUEUE: boolean
  PHONE_NUMBER_PATTERN: string
  NID_NUMBER_PATTERN: string
  USER_NOTIFICATION_DELIVERY_METHOD: string
  INFORMANT_NOTIFICATION_DELIVERY_METHOD: string
  DATE_OF_BIRTH_UNKNOWN: boolean
  INFORMANT_SIGNATURE: boolean
  INFORMANT_SIGNATURE_REQUIRED: boolean
  LOGIN_BACKGROUND: GQLLoginBackgroundInput
}
