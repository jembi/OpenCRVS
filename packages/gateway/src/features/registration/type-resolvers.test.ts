/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */

import { ApolloServer } from 'apollo-server-hapi'

jest.useFakeTimers('modern').setSystemTime(new Date('2023-10-11T07:24:55.108Z'))

import { IUserModelData } from '@gateway/features/user/type-resolvers'
import { getApolloConfig } from '@gateway/graphql/config'
import { generateQueryForType } from '@gateway/graphql/query-generator'
import { readFileSync } from 'fs'
import * as jwt from 'jsonwebtoken'
import {
  BIRTH_BUNDLE,
  DEATH_BUNDLE,
  MARRIAGE_BUNDLE
} from '@opencrvs/commons/fixtures'
import { Context } from '@gateway/graphql/context'
import { getAuthHeader } from '@opencrvs/commons/http'
import {
  getUserRoleFromHistory,
  PractitionerRoleHistory
} from '@opencrvs/commons/types'

const MOCK_TOKEN = jwt.sign(
  { scope: ['validate'] },
  readFileSync('./test/cert.key'),
  {
    subject: '121221',
    algorithm: 'RS256',
    issuer: 'opencrvs:auth-service',
    audience: 'opencrvs:gateway-user'
  }
)

const MOCK_USER: IUserModelData = {
  _id: '6501acbe81245ece9e627bf7',
  name: [
    {
      use: 'en',
      given: ['Kennedy'],
      family: 'Mweene'
    }
  ],
  username: 'd.bulaya',
  identifiers: [],
  email: '',
  device: 'Samsung Galaxy S9',
  emailForNotification: 'kalushabw.alya17@gmail.com',
  mobile: '0934343434',
  systemRole: 'LOCAL_REGISTRAR',
  role: {
    labels: [
      {
        lang: 'en',
        label: 'Local Registrar'
      },
      {
        lang: 'fr',
        label: 'Registraire local'
      }
    ]
  },
  practitionerId: '48455871-1636-46a1-8279-aaa76dec03d4',
  primaryOfficeId: '5faf414c-99e8-47a7-b4d9-d07a02dbc97e',
  scope: ['register', 'performance', 'certify', 'demo'],
  status: 'active',
  auditHistory: []
}

const MOCK_RESPONSES = {
  getUser: MOCK_USER,
  // User's primary office's parent location
  '/Location/c1ddb556-8b22-4182-9a6c-64358c0e96c9': {
    resourceType: 'Location',
    identifier: [
      {
        system: 'http://opencrvs.org/specs/id/internal-id',
        value: 'CRVS_OFFICE_HPGiE9Jjh2r'
      }
    ],
    name: 'HPGiE9Jjh2r',
    alias: ['HPGiE9Jjh2r'],
    status: 'active',
    mode: 'instance',
    partOf: {
      reference: 'Location/0'
    },
    type: {
      coding: [
        {
          system: 'http://opencrvs.org/specs/location-type',
          code: 'CRVS_OFFICE'
        }
      ]
    },
    physicalType: {
      coding: [
        {
          code: 'jdn',
          display: 'Jurisdiction'
        }
      ]
    },
    meta: {
      lastUpdated: '2023-09-13T12:36:08.757+00:00',
      versionId: 'c5e4cf8a-15c6-4bd0-8299-6be0383d5713'
    },
    _transforms: {
      meta: {
        lastUpdated: '2023-09-13T12:36:08.757Z'
      }
    },
    _request: {
      method: 'POST'
    },
    id: 'c1ddb556-8b22-4182-9a6c-64358c0e96c9'
  },
  // User's primary office
  '/Location/5faf414c-99e8-47a7-b4d9-d07a02dbc97e': {
    _id: '6501acb8b91837001d484b6d',
    resourceType: 'Location',
    identifier: [
      {
        system: 'http://opencrvs.org/specs/id/internal-id',
        value: 'CRVS_OFFICE_okQp4uKCz0'
      }
    ],
    name: 'Ilanga District Office',
    alias: ['Ilanga District Office'],
    status: 'active',
    mode: 'instance',
    partOf: {
      reference: 'Location/c1ddb556-8b22-4182-9a6c-64358c0e96c9'
    },
    type: {
      coding: [
        {
          system: 'http://opencrvs.org/specs/location-type',
          code: 'CRVS_OFFICE'
        }
      ]
    },
    physicalType: {
      coding: [
        {
          code: 'bu',
          display: 'Building'
        }
      ]
    },
    meta: {
      lastUpdated: '2023-09-13T12:36:08.757+00:00',
      versionId: 'c5e4cf8a-15c6-4bd0-8299-6be0383d5713'
    },
    _transforms: {
      meta: {
        lastUpdated: '2023-09-13T12:36:08.757Z'
      }
    },
    _request: {
      method: 'POST'
    },
    id: '5faf414c-99e8-47a7-b4d9-d07a02dbc97e'
  },
  '/PractitionerRole?practitioner=511ab59f-54df-4331-8c22-97062a602154': {
    resourceType: 'PractitionerRole',
    practitioner: {
      reference: 'Practitioner/511ab59f-54df-4331-8c22-97062a602154'
    },
    id: '82ef255e-7b49-4e7c-8851-b3f0b440839a'
  },
  '/Practitioner/511ab59f-54df-4331-8c22-97062a602154': {
    _id: '6501acbcb91837001d484b89',
    resourceType: 'Practitioner',
    identifier: [],
    telecom: [
      {
        system: 'phone',
        value: '0934343434'
      },
      {
        system: 'email',
        value: ''
      }
    ],
    name: [
      {
        use: 'en',
        family: 'Mweene',
        given: ['Kennedy']
      }
    ],
    meta: {
      lastUpdated: '2023-09-13T12:36:12.371+00:00',
      versionId: '61171268-0270-4497-87a4-9c464a9d4257'
    },
    id: '511ab59f-54df-4331-8c22-97062a602154'
  }
}

jest.mock('@gateway/features/registration/utils', () => {
  const originalModule = jest.requireActual(
    '@gateway/features/registration/utils'
  )
  return {
    ...originalModule,
    getPresignedUrlFromUri: () => {
      return Promise.resolve('/mock-presigned-url')
    }
  }
})

jest.mock('@gateway/features/user/utils', () => {
  const originalModule = jest.requireActual('@gateway/features/user/utils')
  return {
    ...originalModule,
    getUser: () => Promise.resolve(MOCK_USER)
  }
})

jest.mock('apollo-datasource-rest', () => {
  function fakeHTTPRequest(url: keyof typeof MOCK_RESPONSES) {
    if (MOCK_RESPONSES[url] !== undefined) {
      return MOCK_RESPONSES[url]
    }
    throw new Error(`No mock for url ${url.toString()}`)
  }
  class MockRESTDataSource {
    baseUrl = ''
    get = fakeHTTPRequest
    post = fakeHTTPRequest

    context = {
      record: BIRTH_BUNDLE
    }
  }
  return {
    RESTDataSource: MockRESTDataSource
  }
})

jest.mock('@gateway/workflow/index', () => {
  const originalModule = jest.requireActual('@gateway/workflow/index')
  return {
    ...originalModule,
    fetchRegistrationForDownloading: jest.fn(() =>
      Promise.resolve(BIRTH_BUNDLE)
    )
  }
})

test('running a full aggregated birth FHIR bundle through resolvers produces a BirthRegistration object', async () => {
  const apolloConfig = getApolloConfig()
  const testServer = new ApolloServer(getApolloConfig())

  const query = `query {
    fetchBirthRegistration(id: "") {
    ${generateQueryForType(apolloConfig.schema!, 'BirthRegistration')}
    }
  }
  `

  const response = await testServer.executeOperation(
    {
      query
    },
    {
      request: {
        auth: {
          isAuthenticated: true,
          credentials: {
            scope: ['register', 'performance', 'certify', 'demo']
          }
        },
        headers: {
          authorization: `Bearer ${MOCK_TOKEN}`
        }
      }
    }
  )
  expect(response.data).toMatchSnapshot()
})

test('running a full aggregated death FHIR bundle through resolvers produces a DeathRegistration object', async () => {
  const apolloConfig = getApolloConfig()

  // eslint-disable-next-line @typescript-eslint/no-var-requires
  require('@gateway/workflow/index').fetchRegistrationForDownloading.mockImplementation(
    () => Promise.resolve(DEATH_BUNDLE)
  )
  const testServer = new ApolloServer({
    ...getApolloConfig(),
    context: async ({ request }): Promise<Omit<Context, 'dataSources'>> => {
      return {
        request,
        headers: getAuthHeader(request),
        presignDocumentUrls: true,
        record: DEATH_BUNDLE as any
      }
    }
  })

  const query = `query {
    fetchDeathRegistration(id: "") {
    ${generateQueryForType(apolloConfig.schema!, 'DeathRegistration')}
    }
  }
  `

  const response = await testServer.executeOperation(
    {
      query
    },
    {
      request: {
        auth: {
          isAuthenticated: true,
          credentials: {
            scope: ['register', 'performance', 'certify', 'demo']
          }
        },
        headers: {
          authorization: `Bearer ${MOCK_TOKEN}`
        }
      }
    }
  )
  expect(response.data).toMatchSnapshot()
})
test('running a full aggregated marriage FHIR bundle through resolvers produces a MarriageRegistration object', async () => {
  const apolloConfig = getApolloConfig()

  // eslint-disable-next-line @typescript-eslint/no-var-requires
  require('@gateway/workflow/index').fetchRegistrationForDownloading.mockImplementation(
    () => Promise.resolve(MARRIAGE_BUNDLE)
  )
  const testServer = new ApolloServer({
    ...getApolloConfig(),
    context: async ({ request }): Promise<Omit<Context, 'dataSources'>> => {
      return {
        request,
        headers: getAuthHeader(request),
        presignDocumentUrls: true,
        record: MARRIAGE_BUNDLE
      }
    }
  })

  const query = `query {
    fetchMarriageRegistration(id: "") {
    ${generateQueryForType(apolloConfig.schema!, 'MarriageRegistration')}
    }
  }
  `

  const response = await testServer.executeOperation(
    {
      query
    },
    {
      request: {
        auth: {
          isAuthenticated: true,
          credentials: {
            scope: ['register', 'performance', 'certify', 'demo']
          }
        },
        headers: {
          authorization: `Bearer ${MOCK_TOKEN}`
        }
      }
    }
  )
  expect(response.data).toMatchSnapshot()
})

test('getting role at a specific time from roleHistory', async () => {
  const practitionerRoleHistory = [
    {
      resourceType: 'PractitionerRole',
      practitioner: {
        reference: 'Practitioner/f28e6b7e-30ee-460a-8851-7e71b46c97cd'
      },
      code: [
        {
          coding: [
            {
              system: 'http://opencrvs.org/specs/roles',
              code: 'NATIONAL_REGISTRAR'
            }
          ]
        },
        {
          coding: [
            {
              system: 'http://opencrvs.org/specs/types',
              code: '[{"lang":"en","label":"National Registrar"},{"lang":"fr","label":"Registraire national"}]'
            }
          ]
        }
      ],
      location: [
        { reference: 'Location/1e73d8a7-964d-4ea1-be77-0de1d2ced0a9' }
      ],
      id: 'ef518d5e-28a5-4c6f-9152-da2c28815e9b',
      meta: {
        lastUpdated: '2024-09-02T08:13:50.173+00:00',
        versionId: '4dfc703b-a5f2-4f56-a11e-5827b3160e0f'
      },
      _transforms: { meta: { lastUpdated: '2024-09-02T08:13:50.173Z' } },
      _request: { method: 'PUT' }
    },
    {
      resourceType: 'PractitionerRoleHistory',
      practitioner: {
        reference: 'Practitioner/f28e6b7e-30ee-460a-8851-7e71b46c97cd'
      },
      code: [
        {
          coding: [
            {
              system: 'http://opencrvs.org/specs/roles',
              code: 'LOCAL_REGISTRAR'
            }
          ]
        },
        {
          coding: [
            {
              system: 'http://opencrvs.org/specs/types',
              code: '[{"lang":"en","label":"Local Registrar"},{"lang":"fr","label":"Registraire local"}]'
            }
          ]
        }
      ],
      location: [
        { reference: 'Location/1e73d8a7-964d-4ea1-be77-0de1d2ced0a9' }
      ],
      meta: {
        lastUpdated: '2024-08-30T13:10:48.262+00:00',
        versionId: '95d3b84c-a77c-4634-8033-7802ba4bf5f1'
      },
      _transforms: { meta: { lastUpdated: '2024-08-30T13:10:48.262Z' } },
      _request: { method: 'POST' },
      id: 'ef518d5e-28a5-4c6f-9152-da2c28815e9b'
    },
    {
      resourceType: 'PractitionerRoleHistory',
      practitioner: {
        reference: 'Practitioner/f28e6b7e-30ee-460a-8851-7e71b46c97cd'
      },
      code: [
        {
          coding: [
            {
              system: 'http://opencrvs.org/specs/roles',
              code: 'NATIONAL_REGISTRAR'
            }
          ]
        },
        {
          coding: [
            {
              system: 'http://opencrvs.org/specs/types',
              code: '[{"lang":"en","label":"National Registrar"},{"lang":"fr","label":"Registraire national"}]'
            }
          ]
        }
      ],
      location: [
        { reference: 'Location/1e73d8a7-964d-4ea1-be77-0de1d2ced0a9' }
      ],
      id: 'ef518d5e-28a5-4c6f-9152-da2c28815e9b',
      meta: {
        lastUpdated: '2024-08-30T13:14:16.965+00:00',
        versionId: '05e67e48-9774-43c4-9136-53653d8577b4'
      },
      _transforms: { meta: { lastUpdated: '2024-08-30T13:14:16.965Z' } },
      _request: { method: 'PUT' }
    },
    {
      resourceType: 'PractitionerRoleHistory',
      practitioner: {
        reference: 'Practitioner/f28e6b7e-30ee-460a-8851-7e71b46c97cd'
      },
      code: [
        {
          coding: [
            {
              system: 'http://opencrvs.org/specs/roles',
              code: 'LOCAL_SYSTEM_ADMIN'
            }
          ]
        },
        {
          coding: [
            {
              system: 'http://opencrvs.org/specs/types',
              code: '[{"lang":"en","label":"Local System Admin"},{"lang":"fr","label":"Administrateur système local"}]'
            }
          ]
        }
      ],
      location: [
        { reference: 'Location/1e73d8a7-964d-4ea1-be77-0de1d2ced0a9' }
      ],
      id: 'ef518d5e-28a5-4c6f-9152-da2c28815e9b',
      meta: {
        lastUpdated: '2024-09-02T08:12:52.860+00:00',
        versionId: '70690251-e1cf-4196-ae9a-9b579edd6e10'
      },
      _transforms: { meta: { lastUpdated: '2024-09-02T08:12:52.860Z' } },
      _request: { method: 'PUT' }
    }
  ] as unknown as PractitionerRoleHistory[]

  /*
    2024-09-02T08:13:50.173+00:00 NATIONAL_REGISTRAR
    2024-08-30T13:10:48.262+00:00 LOCAL_REGISTRAR
    2024-08-30T13:14:16.965+00:00 NATIONAL_REGISTRAR *
    2024-09-02T08:12:52.860+00:00 LOCAL_SYSTEM_ADMIN
  */

  const lastModified = '2024-08-30T13:14:33.704Z'
  const expectedRole =
    '[{"lang":"en","label":"National Registrar"},{"lang":"fr","label":"Registraire national"}]'
  const expectedSystemRole = 'NATIONAL_REGISTRAR'

  const { role, systemRole } = getUserRoleFromHistory(
    practitionerRoleHistory,
    lastModified
  )

  expect(role).toEqual(expectedRole)
  expect(systemRole).toEqual(expectedSystemRole)
})
