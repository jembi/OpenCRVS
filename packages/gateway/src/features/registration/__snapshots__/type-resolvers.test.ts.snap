// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`running a full aggregated birth FHIR bundle through resolvers produces a BirthRegistration object 1`] = `
Object {
  "fetchBirthRegistration": Object {
    "_fhirIDMap": Object {
      "composition": "09f39e17-ca23-4905-9ab3-f3649c4e3162",
      "encounter": "e7ed42ae-bd54-4d7c-ad9d-0af7506d1d64",
      "eventLocation": "565d957a-109a-4bf3-a3af-f0ad52c719cd",
      "observation": Object {
        "attendantAtBirth": "942901b9-e03f-4209-953d-fd094e5e8231",
        "birthType": "78e42deb-c50e-43c8-853f-71ccc6398512",
        "weightAtBirth": "d47d4a55-bf18-4704-ac3a-0066e3ecff6f",
      },
      "questionnaireResponse": undefined,
    },
    "attendantAtBirth": "PHYSICIAN",
    "birthType": "SINGLE",
    "child": Object {
      "_fhirID": null,
      "address": null,
      "age": null,
      "ageOfIndividualInYears": null,
      "birthDate": "2023-07-11",
      "dateOfMarriage": null,
      "deceased": Object {
        "deathDate": null,
        "deceased": null,
      },
      "detailsExist": true,
      "educationalAttainment": null,
      "exactDateOfBirthUnknown": null,
      "gender": "female",
      "id": "dfac3127-963e-4ef2-b875-3123e08da3d9",
      "identifier": Array [
        Object {
          "fieldsModifiedByIdentity": null,
          "id": "2023B3VUXES",
          "otherType": null,
          "type": "BIRTH_REGISTRATION_NUMBER",
        },
        Object {
          "fieldsModifiedByIdentity": null,
          "id": "",
          "otherType": null,
          "type": "BIRTH_CONFIGURABLE_IDENTIFIER_1",
        },
        Object {
          "fieldsModifiedByIdentity": null,
          "id": "",
          "otherType": null,
          "type": "BIRTH_CONFIGURABLE_IDENTIFIER_2",
        },
        Object {
          "fieldsModifiedByIdentity": null,
          "id": "",
          "otherType": null,
          "type": "BIRTH_CONFIGURABLE_IDENTIFIER_3",
        },
      ],
      "maritalStatus": null,
      "multipleBirth": null,
      "name": Array [
        Object {
          "familyName": "Baumbach",
          "firstNames": "Isadore",
          "marriedLastName": null,
          "use": "en",
        },
      ],
      "nationality": null,
      "occupation": null,
      "photo": null,
      "reasonNotApplying": null,
      "telecom": null,
    },
    "childrenBornAliveToMother": null,
    "createdAt": "2023-08-11T05:26:26.000Z",
    "eventLocation": Object {
      "_fhirID": null,
      "address": null,
      "altitude": null,
      "description": null,
      "geoData": null,
      "id": "565d957a-109a-4bf3-a3af-f0ad52c719cd",
      "latitude": null,
      "longitude": null,
      "name": "Keembe Rural Health Centre",
      "partOf": "Location/e66643ac-9ea9-4314-b842-f4fb3ad9e83a",
      "status": "active",
      "telecom": null,
      "type": "HEALTH_FACILITY",
    },
    "father": Object {
      "_fhirID": null,
      "address": Array [
        Object {
          "city": "South Jamelmouth",
          "country": "FAR",
          "district": "e66643ac-9ea9-4314-b842-f4fb3ad9e83a",
          "districtName": "Ibombo",
          "from": null,
          "line": Array [
            "12",
            "Usual Street",
            "Usual Residental Area",
            "",
            "",
            "URBAN",
          ],
          "lineName": Array [
            "12",
            "Usual Street",
            "Usual Residental Area",
            "",
            "",
            "URBAN",
          ],
          "partOf": "e66643ac-9ea9-4314-b842-f4fb3ad9e83a",
          "postalCode": "34422-6447",
          "state": "1cfe40fa-7b43-4c1e-aa05-4281e5122d9b",
          "stateName": "Central",
          "text": null,
          "to": null,
          "type": "PRIMARY_ADDRESS",
          "use": null,
        },
      ],
      "age": null,
      "ageOfIndividualInYears": null,
      "birthDate": "2003-07-11",
      "dateOfMarriage": "2021-07-11",
      "deceased": Object {
        "deathDate": null,
        "deceased": null,
      },
      "detailsExist": true,
      "educationalAttainment": "POST_SECONDARY_ISCED_4",
      "exactDateOfBirthUnknown": null,
      "gender": null,
      "id": "3ee05cf9-a550-4d7f-abde-54313f7bd41a",
      "identifier": Array [
        Object {
          "fieldsModifiedByIdentity": null,
          "id": "8762475961",
          "otherType": null,
          "type": "NATIONAL_ID",
        },
      ],
      "maritalStatus": "MARRIED",
      "multipleBirth": null,
      "name": Array [
        Object {
          "familyName": "Neil",
          "firstNames": "Baumbach",
          "marriedLastName": null,
          "use": "en",
        },
      ],
      "nationality": Array [
        "FAR",
      ],
      "occupation": "Bookkeeper",
      "photo": null,
      "reasonNotApplying": null,
      "telecom": null,
    },
    "foetalDeathsToMother": null,
    "history": Array [
      Object {
        "certificates": Array [
          null,
        ],
        "comments": Array [],
        "date": "2023-10-02T13:51:55.645+00:00",
        "dhis2Notification": false,
        "duplicateOf": null,
        "hasShowedVerifiedDocument": false,
        "input": Array [],
        "ipAddress": null,
        "location": Object {
          "_fhirID": null,
          "address": null,
          "altitude": null,
          "description": "oEBf29y8JP8",
          "geoData": null,
          "id": "e66643ac-9ea9-4314-b842-f4fb3ad9e83a",
          "latitude": null,
          "longitude": null,
          "name": "Ibombo",
          "partOf": "Location/1cfe40fa-7b43-4c1e-aa05-4281e5122d9b",
          "status": "active",
          "telecom": null,
          "type": "ADMIN_STRUCTURE",
        },
        "noSupportingDocumentationRequired": false,
        "note": "",
        "office": Object {
          "_fhirID": null,
          "address": null,
          "altitude": null,
          "description": null,
          "geoData": null,
          "id": "e9e1b362-27c9-4ce1-82ad-57fe9d5650e4",
          "latitude": null,
          "longitude": null,
          "name": "Ibombo District Office",
          "partOf": "Location/e66643ac-9ea9-4314-b842-f4fb3ad9e83a",
          "status": "active",
          "telecom": null,
          "type": "CRVS_OFFICE",
        },
        "otherReason": "",
        "output": Array [],
        "payment": null,
        "reason": null,
        "requester": "",
        "requesterOther": "",
        "signature": null,
        "statusReason": null,
        "system": null,
        "user": null,
      },
    ],
    "id": "09f39e17-ca23-4905-9ab3-f3649c4e3162",
    "informant": Object {
      "_fhirID": null,
      "_fhirIDPatient": "cd5693ed-dd33-4c3a-a9fc-7479b132a4ea",
      "address": Array [
        Object {
          "city": "South Jamelmouth",
          "country": "FAR",
          "district": "e66643ac-9ea9-4314-b842-f4fb3ad9e83a",
          "districtName": "Ibombo",
          "from": null,
          "line": Array [
            "12",
            "Usual Street",
            "Usual Residental Area",
            "",
            "",
            "URBAN",
            "12",
            "Usual Street",
            "Usual Residental Area",
            "",
            "",
            "URBAN",
          ],
          "lineName": Array [
            "12",
            "Usual Street",
            "Usual Residental Area",
            "",
            "",
            "URBAN",
            "12",
            "Usual Street",
            "Usual Residental Area",
            "",
            "",
            "URBAN",
          ],
          "partOf": "e66643ac-9ea9-4314-b842-f4fb3ad9e83a",
          "postalCode": "34422-6447",
          "state": "1cfe40fa-7b43-4c1e-aa05-4281e5122d9b",
          "stateName": "Central",
          "text": null,
          "to": null,
          "type": "PRIMARY_ADDRESS",
          "use": null,
        },
      ],
      "affidavit": null,
      "age": null,
      "ageOfIndividualInYears": null,
      "birthDate": "2003-07-11",
      "dateOfMarriage": "2021-07-11",
      "deceased": Object {
        "deathDate": null,
        "deceased": null,
      },
      "detailsExist": true,
      "educationalAttainment": "POST_SECONDARY_ISCED_4",
      "exactDateOfBirthUnknown": null,
      "gender": null,
      "id": "a0fd7881-33e6-4751-a346-78f820aa3dbb",
      "identifier": Array [
        Object {
          "fieldsModifiedByIdentity": null,
          "id": "2253499797",
          "otherType": null,
          "type": "NATIONAL_ID",
        },
      ],
      "maritalStatus": "MARRIED",
      "multipleBirth": 2,
      "name": Array [
        Object {
          "familyName": "Baumbach",
          "firstNames": "Sophia",
          "marriedLastName": null,
          "use": "en",
        },
      ],
      "nationality": Array [
        "FAR",
      ],
      "occupation": "Farmer",
      "otherRelationship": null,
      "photo": null,
      "reasonNotApplying": null,
      "relationship": "MOTHER",
      "telecom": null,
    },
    "lastPreviousLiveBirth": null,
    "mother": Object {
      "_fhirID": null,
      "address": Array [
        Object {
          "city": "South Jamelmouth",
          "country": "FAR",
          "district": "e66643ac-9ea9-4314-b842-f4fb3ad9e83a",
          "districtName": "Ibombo",
          "from": null,
          "line": Array [
            "12",
            "Usual Street",
            "Usual Residental Area",
            "",
            "",
            "URBAN",
            "12",
            "Usual Street",
            "Usual Residental Area",
            "",
            "",
            "URBAN",
          ],
          "lineName": Array [
            "12",
            "Usual Street",
            "Usual Residental Area",
            "",
            "",
            "URBAN",
            "12",
            "Usual Street",
            "Usual Residental Area",
            "",
            "",
            "URBAN",
          ],
          "partOf": "e66643ac-9ea9-4314-b842-f4fb3ad9e83a",
          "postalCode": "34422-6447",
          "state": "1cfe40fa-7b43-4c1e-aa05-4281e5122d9b",
          "stateName": "Central",
          "text": null,
          "to": null,
          "type": "PRIMARY_ADDRESS",
          "use": null,
        },
      ],
      "age": null,
      "ageOfIndividualInYears": null,
      "birthDate": "2003-07-11",
      "dateOfMarriage": "2021-07-11",
      "deceased": Object {
        "deathDate": null,
        "deceased": null,
      },
      "detailsExist": true,
      "educationalAttainment": "POST_SECONDARY_ISCED_4",
      "exactDateOfBirthUnknown": null,
      "gender": null,
      "id": "cd5693ed-dd33-4c3a-a9fc-7479b132a4ea",
      "identifier": Array [
        Object {
          "fieldsModifiedByIdentity": null,
          "id": "2253499797",
          "otherType": null,
          "type": "NATIONAL_ID",
        },
      ],
      "maritalStatus": "MARRIED",
      "multipleBirth": 2,
      "name": Array [
        Object {
          "familyName": "Baumbach",
          "firstNames": "Sophia",
          "marriedLastName": null,
          "use": "en",
        },
      ],
      "nationality": Array [
        "FAR",
      ],
      "occupation": "Farmer",
      "photo": null,
      "reasonNotApplying": null,
      "telecom": Array [
        Object {
          "system": "phone",
          "use": null,
          "value": "+260711384538",
        },
      ],
    },
    "otherAttendantAtBirth": null,
    "questionnaire": null,
    "registration": Object {
      "_fhirID": null,
      "assignment": Object {
        "avatarURL": "/mock-presigned-url",
        "firstName": "Derrick",
        "lastName": "Bulaya",
        "officeName": "Ibombo District Office",
        "userId": "6501acbe81245ece9e627bf7",
      },
      "attachments": null,
      "book": null,
      "brideSignature": null,
      "brideSignatureURI": null,
      "certificates": Array [
        null,
      ],
      "contact": "MOTHER",
      "contactEmail": "Sophia.Baumbach95@gmail.com",
      "contactPhoneNumber": "+260711384538",
      "contactRelationship": null,
      "draftId": null,
      "duplicates": null,
      "groomSignature": null,
      "groomSignatureURI": null,
      "id": "358c0c75-9855-4812-b560-c1b5fbf48e5a",
      "inCompleteFields": null,
      "informantType": "MOTHER",
      "informantsSignature": null,
      "informantsSignatureURI": null,
      "mosipAid": null,
      "otherInformantType": null,
      "page": null,
      "paperFormID": null,
      "registrationNumber": "2023B3VUXES",
      "status": Array [
        Object {
          "comments": null,
          "id": "358c0c75-9855-4812-b560-c1b5fbf48e5a",
          "location": Object {
            "_fhirID": null,
            "address": null,
            "altitude": null,
            "description": "oEBf29y8JP8",
            "geoData": null,
            "id": "e66643ac-9ea9-4314-b842-f4fb3ad9e83a",
            "latitude": null,
            "longitude": null,
            "name": "Ibombo",
            "partOf": "Location/1cfe40fa-7b43-4c1e-aa05-4281e5122d9b",
            "status": "active",
            "telecom": null,
            "type": "ADMIN_STRUCTURE",
          },
          "office": Object {
            "_fhirID": null,
            "address": null,
            "altitude": null,
            "description": null,
            "geoData": null,
            "id": "e9e1b362-27c9-4ce1-82ad-57fe9d5650e4",
            "latitude": null,
            "longitude": null,
            "name": "Ibombo District Office",
            "partOf": "Location/e66643ac-9ea9-4314-b842-f4fb3ad9e83a",
            "status": "active",
            "telecom": null,
            "type": "CRVS_OFFICE",
          },
          "reason": null,
          "timeLogged": null,
          "timestamp": "2023-10-02T13:51:55.410Z",
          "user": null,
        },
      ],
      "trackingId": "B3VUXES",
      "witnessOneSignature": null,
      "witnessOneSignatureURI": null,
      "witnessTwoSignature": null,
      "witnessTwoSignatureURI": null,
    },
    "updatedAt": null,
    "weightAtBirth": 4.1,
  },
}
`;
